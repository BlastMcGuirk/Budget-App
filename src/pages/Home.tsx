import { NativeStackScreenProps } from '@react-navigation/native-stack';
import React, { useEffect } from 'react';
import { ScrollView, StyleSheet, Text } from 'react-native';
import { RootStackParamList } from '../App';
import BudgetSummaries from '../components/BudgetSummaries';
import SpendingList from '../components/SpendingList';
import { RootState, useAppDispatch } from '../redux/store';
import { useSelector } from 'react-redux';
import { loadData } from '../redux/features/budget-slice';
import { DateNavigator } from '../components/DateNavigator';
import { Budget } from '../interfaces/Budget';

type Props = NativeStackScreenProps<RootStackParamList, 'Home'>;

export default function Home(props: Props) {
    const {budgets, loading} = useSelector((state: RootState) => state.budgets);

    const dispatch = useAppDispatch();

    useEffect(() => {
        dispatch(loadData());
    }, []);

    const navigateToBudgetDetails = function(budgetId: number) {
        props.navigation.navigate("BudgetDetails", {budgetId: budgetId});
    }

    const navigateToNew = function(budget: Budget) {
        props.navigation.navigate("NewEntry", {budget, returnTo: 'Home', returnProps: undefined});
    }

    const navigateToItemDetails = function(budgetId: number, itemId: number) {
        props.navigation.navigate("ItemDetails", { budgetId, itemId });
    }

    return (
        <ScrollView contentContainerStyle={styles.container}>
            {loading && <Text>Loading...</Text>}
            {!loading && <>
                <DateNavigator />
                <BudgetSummaries />
                {budgets.map(budget => {
                    return <SpendingList 
                        key={budget.id}
                        budget={budget}
                        items={budget.items}
                        onNew={navigateToNew}
                        onNavigateBudget={navigateToBudgetDetails}
                        onNavigateItem={navigateToItemDetails} />
                })}
            </>}
        </ScrollView>
    )
}

const styles = StyleSheet.create({
    container: {
        backgroundColor: '#fff',
        alignItems: 'center',
    },
});
