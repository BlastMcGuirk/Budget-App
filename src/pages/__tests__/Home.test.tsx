import { NativeStackScreenProps } from '@react-navigation/native-stack';
import renderer from 'react-test-renderer';
import Home from "../Home";

describe("Home Page Tests", () => {

    it("Renders correctly", () => {
        const props: any = {};
        const tree = renderer.
            create(<Home {...props} />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

});