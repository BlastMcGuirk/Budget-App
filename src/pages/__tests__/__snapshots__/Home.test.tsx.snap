// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Page Tests Renders correctly 1`] = `
<RCTScrollView
  contentContainerStyle={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  <View>
    <View
      style={
        Object {
          "alignItems": "baseline",
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
            "padding": 20,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 24,
            }
          }
        >
          NEEDS
        </Text>
        <Text
          style={
            Array [
              Object {
                "fontSize": 36,
              },
              Object {
                "color": "green",
              },
            ]
          }
        >
          $
          1234.56
        </Text>
        <Text
          style={
            Object {
              "fontSize": 16,
            }
          }
        >
          / $
          1500
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
            "padding": 20,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 24,
            }
          }
        >
          WANTS
        </Text>
        <Text
          style={
            Array [
              Object {
                "fontSize": 36,
              },
              Object {
                "color": "red",
              },
            ]
          }
        >
          $
          -123.45
        </Text>
        <Text
          style={
            Object {
              "fontSize": 16,
            }
          }
        >
          / $
          500
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "borderTopWidth": 2,
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "borderBottomColor": "gray",
            "borderBottomWidth": 2,
            "padding": 8,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "baseline",
              "display": "flex",
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 24,
              }
            }
          >
            Needs
          </Text>
          <Text
            onPress={[Function]}
            style={
              Object {
                "fontSize": 24,
              }
            }
          >
            + New
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "borderBottomColor": "gray",
            "borderBottomWidth": 1,
            "padding": 16,
            "width": "100%",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "baseline",
              "display": "flex",
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 24,
              }
            }
          >
            Rent
          </Text>
          <Text
            style={
              Object {
                "fontSize": 24,
              }
            }
          >
            $
            852.25
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "baseline",
              "display": "flex",
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 16,
              }
            }
          >
            Bills
          </Text>
          <Text
            style={
              Object {
                "fontSize": 16,
              }
            }
          >
            8/4/2022
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "borderBottomColor": "gray",
            "borderBottomWidth": 1,
            "padding": 16,
            "width": "100%",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "baseline",
              "display": "flex",
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 24,
              }
            }
          >
            Groceries
          </Text>
          <Text
            style={
              Object {
                "fontSize": 24,
              }
            }
          >
            $
            130.44
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "baseline",
              "display": "flex",
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 16,
              }
            }
          >
            Groceries
          </Text>
          <Text
            style={
              Object {
                "fontSize": 16,
              }
            }
          >
            8/7/2022
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "borderBottomColor": "gray",
            "borderBottomWidth": 1,
            "padding": 16,
            "width": "100%",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "baseline",
              "display": "flex",
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 24,
              }
            }
          >
            Rent1
          </Text>
          <Text
            style={
              Object {
                "fontSize": 24,
              }
            }
          >
            $
            852.25
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "baseline",
              "display": "flex",
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 16,
              }
            }
          >
            Bills
          </Text>
          <Text
            style={
              Object {
                "fontSize": 16,
              }
            }
          >
            8/4/2022
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "borderBottomWidth": 2,
            "padding": 8,
          }
        }
      >
        <Text
          onPress={[Function]}
          style={
            Object {
              "fontSize": 20,
            }
          }
        >
          See All
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "borderTopWidth": 2,
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "borderBottomColor": "gray",
            "borderBottomWidth": 2,
            "padding": 8,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "baseline",
              "display": "flex",
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 24,
              }
            }
          >
            Wants
          </Text>
          <Text
            onPress={[Function]}
            style={
              Object {
                "fontSize": 24,
              }
            }
          >
            + New
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "borderBottomColor": "gray",
            "borderBottomWidth": 1,
            "padding": 16,
            "width": "100%",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "baseline",
              "display": "flex",
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 24,
              }
            }
          >
            Date Night
          </Text>
          <Text
            style={
              Object {
                "fontSize": 24,
              }
            }
          >
            $
            56.00
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "baseline",
              "display": "flex",
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 16,
              }
            }
          />
          <Text
            style={
              Object {
                "fontSize": 16,
              }
            }
          >
            8/12/2022
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "borderBottomColor": "gray",
            "borderBottomWidth": 1,
            "padding": 16,
            "width": "100%",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "baseline",
              "display": "flex",
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 24,
              }
            }
          >
            Rent3
          </Text>
          <Text
            style={
              Object {
                "fontSize": 24,
              }
            }
          >
            $
            852.25
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "baseline",
              "display": "flex",
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 16,
              }
            }
          >
            Bills
          </Text>
          <Text
            style={
              Object {
                "fontSize": 16,
              }
            }
          >
            8/4/2022
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "borderBottomColor": "gray",
            "borderBottomWidth": 1,
            "padding": 16,
            "width": "100%",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "baseline",
              "display": "flex",
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 24,
              }
            }
          >
            Rent4
          </Text>
          <Text
            style={
              Object {
                "fontSize": 24,
              }
            }
          >
            $
            852.25
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "baseline",
              "display": "flex",
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 16,
              }
            }
          >
            Bills
          </Text>
          <Text
            style={
              Object {
                "fontSize": 16,
              }
            }
          >
            8/4/2022
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "borderBottomWidth": 2,
            "padding": 8,
          }
        }
      >
        <Text
          onPress={[Function]}
          style={
            Object {
              "fontSize": 20,
            }
          }
        >
          See All
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
